counsel.el

```diff
@@ -1438,22 +1438,42 @@ This function should set `ivy--old-re'."
 
 (defun counsel-git-grep-action (x)
   "Go to occurrence X in current Git repository."
-  (when (string-match "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'" x)
+  (when (string-match (counsel-git-grep-match-regex) x)
     (let ((file-name (match-string-no-properties 1 x))
           (line-number (match-string-no-properties 2 x)))
       (find-file (expand-file-name
                   file-name
                   (ivy-state-directory ivy-last)))
-      (goto-char (point-min))
-      (forward-line (1- (string-to-number line-number)))
-      (when (re-search-forward (ivy--regex ivy-text t) (line-end-position) t)
-        (when swiper-goto-start-of-match
-          (goto-char (match-beginning 0))))
-      (swiper--ensure-visible)
-      (run-hooks 'counsel-grep-post-action-hook)
-      (unless (eq ivy-exit 'done)
-        (swiper--cleanup)
-        (swiper--add-overlays (ivy--regex ivy-text))))))
+      (counsel-git-grep-go-to-location line-number))))
+
+(defun counsel-git-grep-match-regex ()
+  "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'")
```

Comment:
As mentioned in another comment, I don't think definining this particular function this way is that helpful.  Two more comments:
- Internal functions should have a double hyphen in their name: `counsel--...`
- All functions should have a docstring.  Prior to Emacs 30, this string would also have been interpreted as the function's docstring, which we'd like to avoid.